solr/solrj/src/java/org/apache/solr/common/cloud/Replica.java

```diff
@@ -129,11 +143,23 @@ public Replica(String name, Map<String,Object> propMap, String collection, Strin
     Objects.requireNonNull(this.nodeName, "'node_name' must not be null");
     Objects.requireNonNull(this.core, "'core' must not be null");
     Objects.requireNonNull(this.type, "'type' must not be null");
-    if (propMap.get(ZkStateReader.STATE_PROP) != null) {
-      this.state = State.getState((String) propMap.get(ZkStateReader.STATE_PROP));
+    ClusterState.getReplicaStatesProvider().get().ifPresent(it -> {
+      log.debug("A replica  {} state fetched from per-replica state", name);
```

Comment:
nit: I think this would be confusing, people might think it means that the state fetched was "A" for Active. Would log "Replica..."